app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt

```diff
@@ -20,45 +20,61 @@
 
 package com.wire.android.ui.home.sync
 
+import androidx.appcompat.app.AppCompatActivity
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
+import androidx.core.app.ShareCompat
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.wire.android.appLogger
+import com.wire.android.di.KaliumCoreLogic
 import com.wire.android.ui.home.FeatureFlagState
+import com.wire.kalium.logic.CoreLogic
 import com.wire.kalium.logic.data.sync.SyncState
-import com.wire.kalium.logic.feature.user.ObserveFileSharingStatusUseCase
-import com.wire.kalium.logic.sync.ObserveSyncStateUseCase
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.auth.AccountInfo
+import com.wire.kalium.logic.feature.session.CurrentSessionResult
+import com.wire.kalium.logic.feature.session.CurrentSessionUseCase
+import com.wire.kalium.logic.feature.session.GetAllSessionsResult
+import com.wire.kalium.logic.feature.session.GetSessionsUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
 import javax.inject.Inject
 
 @HiltViewModel
 class FeatureFlagNotificationViewModel @Inject constructor(
-    private val observeSyncState: ObserveSyncStateUseCase,
-    private val observeFileSharingStatusUseCase: ObserveFileSharingStatusUseCase
+    @KaliumCoreLogic private val coreLogic: CoreLogic,
+    private val getSessions: GetSessionsUseCase,
+    private val currentSessionUseCase: CurrentSessionUseCase
 ) : ViewModel() {
 
     var featureFlagState by mutableStateOf(FeatureFlagState())
         private set
 
-    init {
+    fun loadSync() {
```

Comment:
loadSync ?

Reply:
This part was added by Big Mo. I renamed to `loadInitialSync` and added some extended comment.